prov/shm/src/smr_rma.c

```diff
@@ -156,20 +157,33 @@ ssize_t smr_generic_rma(struct smr_ep *ep, const struct iovec *iov,
 	total_len = ofi_total_iov_len(iov, iov_count);
 
 	smr_generic_format(cmd, peer_id, op, 0, data, op_flags);
-	if (!smr_env.disable_cma && (total_len > SMR_INJECT_SIZE ||
-	    op != ofi_op_write || op_flags & FI_DELIVERY_COMPLETE)) {
+	if (total_len > SMR_INJECT_SIZE || (!smr_env.disable_cma &&
+	    (op != ofi_op_write || op_flags & FI_DELIVERY_COMPLETE))) {
```

Comment:
I don't think the reorganization helps here and I'm not sure it's correct. I can't tell. 

I think we also need to assess whether we still want to have the DISABLE_CMA variable and, if so, what effect it's going to have. I think if we do keep it, we need to move it out of the protocol and into the cma_check where we force cma_cap to OFF if disable_cma is on. But I don't really think the environment variable is very useful here anymore. It was mostly a workaround until we had an alternate large message protocol/proper CMA detection. Right now, I think we will have an issue here because if we disable cma with the environment variable, this check will fail and we will go into the inject call but then the message will be too large. I think we should remove it entirely.

Reply:
Yeah, it makes sense, as we already have a peer-to-peer cma detection.
I would remove the env. variable, smr_check_ptrace_scope() entirely. 
Thank you.

Reply:
The change is in commit 32b08cb